x-pack/legacy/plugins/siem/public/hooks/api/api.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+
+import { useKibanaUiSetting } from '../../lib/settings/use_kibana_ui_setting';
+import { DEFAULT_KBN_VERSION } from '../../../common/constants';
+import * as i18n from '../translations';
+import { parseJsonFromBody, ToasterErrors } from '../../components/ml/api/throw_if_not_ok';
+import { IndexPatternResponse, IndexPatternSavedObject } from '../types';
+
+const emptyIndexPattern: IndexPatternSavedObject[] = [];
+
+/**
+ * Fetches Configured Index Patterns from the Kibana saved objects API
+ *
+ * TODO: Refactor to context provider: https://github.com/elastic/siem-team/issues/448
+ *
+ * @param headers
+ * @param signal
+ */
+export const getIndexPatterns = async (
+  signal: AbortSignal,
+  headers?: Record<string, string | undefined>
+): Promise<IndexPatternSavedObject[]> => {
+  const [kbnVersion] = useKibanaUiSetting(DEFAULT_KBN_VERSION);
+  const response = await fetch(
+    `${chrome.getBasePath()}/api/saved_objects/_find?type=index-pattern&fields=title&fields=type&per_page=10000`,
+    {
+      method: 'GET',
+      credentials: 'same-origin',
+      headers: {
+        'content-type': 'application/json',
+        'kbn-xsrf': kbnVersion,
+        'kbn-version': kbnVersion,
+        'kbn-system-api': 'true',
+        ...headers,
+      },
+      signal,
+    }
+  );
+  await throwIfNotOk(response);
+  const results: IndexPatternResponse = await response.json();
+
+  if (results.saved_objects && Array.isArray(results.saved_objects)) {
```

Comment:
That's interesting, when is `results.saved_objects` not an array?

Reading through I almost want a fun fact if there is one here ;-)

```ts
//fun fact:
```

Reply:
So had to double check since this was old code that was just getting moved, but with all Kibana index patterns deleted it still returns an array. 

e.g:

``` js
{"page":1,"per_page":10000,"total":0,"saved_objects":[]}
```

So that check is redundant, but who doesn't like some resilient code?